### PR TITLE
xds: Move bootstrap config to internal/xds

### PR DESCRIPTION
**Pull request by zasweq**
This PR moves bootstrap config to internal/xds. This allows CSM Plugin Option to call bootstrap config parsing to parse the out the mesh id from the node proto's id.

In the future, we could perhaps make the bootstrap config a singleton (and wrap creation function with something like a sync.OnceFunc).

RELEASE NOTES: N/A

### CODE REVIEW

File: internal/xds/bootstrap/credentials.go
Comment (dfawley) [inline internal/xds/bootstrap/credentials.go:1]: This is fine...

But do you think maybe this logic (and this file) would be even better in `grpc/xds/bootstrap`?  (And then blank import that package from this package's `bootstrap.go`?

I'm OK either way, but would lean toward moving it in with the registry if you can't think of any reason it's better off (or needs to be?) here.  These are essentially "default supported creds", and if it's possible for a user to import the registry but not get these creds in it, that seems weird.

**Review by dfawley — APPROVED** [review]

File: internal/xds/bootstrap/credentials.go
Comment (zasweq) [reply internal/xds/bootstrap/credentials.go:1]: I'm fine with either. I'll move it.

Comment (zasweq) [reply internal/xds/bootstrap/credentials.go:1]: No need to blank import, the package is already imported :).

**Review by dfawley — APPROVED** [review]

File: internal/xds/bootstrap/credentials.go
Comment (dfawley) [reply internal/xds/bootstrap/credentials.go:1]: Oh yeah, it does `Get`s

**Pull request merged**